conans/client/build/cmake.py

```diff
@@ -114,15 +115,23 @@ def flags(self):
     def is_multi_configuration(self):
         return is_multi_configuration(self.generator)
 
+    @property
+    def is_toolset_supported(self):
+        return is_toolset_supported(self.generator)
+
+    @property
+    def is_generator_platform_supported(self):
+        return is_generator_platform_supported(self.generator)
+
     @property
     def command_line(self):
         args = ['-G "%s"' % self.generator] if self.generator else []
-        if self.generator_platform:
+        if self.generator_platform and self.is_generator_platform_supported:
```

Comment:
Silently skipping this, and not using a specified toolset, which produces a different binary, is an error that should raise and alert the users of a misconfiguration in their profiles.

Reply:
I've changed the code to raise exception in case of unsupported combinations